### PR TITLE
AVX-474: Add view modes to Pixy Viewer

### PR DESCRIPTION
**Pull request by dat-tran-matternet**
This change adds a menu option for the user to select three view modes:

  - Block:  Block (aka blob) detection mode
  - Video:  Video mode
  - Merge:  Merge of block and video mode

To support video mode, the handling of images transferred over USB is wrapped to
expose the images to Python.

Support for Windows is added by upgrading to Python3 and removing boost
dependency.

### CODE REVIEW

File: src/host/libpixyusb/src/pixyinterpreter.cpp [hunk 1]
```diff
@@ -162,7 +175,7 @@ void PixyInterpreter::interpreter_thread()
 
     // Mutual exclusion for receiver_ object (Unlock) //
     chirp_access_mutex_.unlock();
-    usleep(15000); // wait for 15ms, ie give time for 
+    std::this_thread::sleep_for(std::chrono::milliseconds(15)); // wait for 15ms, ie give time for
   }
```
Comment (zalice) [inline src/host/libpixyusb/src/pixyinterpreter.cpp:166]: This is pre-existing: the comment is incomplete. Could you add words or remove the second half of it?

**Review by zalice — APPROVED** [review]

File: scripts/build_libpixyusb_swig.bat [hunk 2]
```diff
@@ -0,0 +1,47 @@
+@ECHO off
```
Comment (rabad-matternet) [inline scripts/build_libpixyusb_swig.bat:1]: Is this a Windows batch file?

File: src/host/libpixyusb/src/pixy.cpp [hunk 3]
```diff
@@ -90,7 +90,7 @@ extern "C"
 
     return_value = interpreter.init();
 
-    if(return_value == 0) 
+    if(return_value == 0)
     {
```
Comment (rabad-matternet) [inline src/host/libpixyusb/src/pixy.cpp]: It looks like we modified a third-party source file as our own now.  If it is indeed our own file, we should further modify to fit our coding style.

This comment applies to other files in this commit as well.

File: scripts/build_libpixyusb_swig.bat (hunk 2 above)
Comment (dat-tran-matternet) [reply scripts/build_libpixyusb_swig.bat:1]: Correct. This is to support the Windows end-users.

File: src/host/libpixyusb/src/pixyinterpreter.cpp (hunk 1 above)
Comment (dat-tran-matternet) [reply src/host/libpixyusb/src/pixyinterpreter.cpp:166]: done

File: src/host/libpixyusb/src/pixy.cpp (hunk 3 above)
Comment (dat-tran-matternet) [reply src/host/libpixyusb/src/pixy.cpp]: done

**Review by zalice — APPROVED** [review]

**Review by AydenChubbic — APPROVED** [review]

**Pull request merged**